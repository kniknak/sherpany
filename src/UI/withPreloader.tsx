import React, { Suspense } from "react";
import { Preloader } from "./Preloader";

export const withPreloader = <TProps extends object = {}>(Component: React.ComponentType<TProps>): React.ComponentType<TProps> => (
    (props) => (
        <Suspense fallback={<Preloader>loading...</Preloader>}>
            <Component {...props}/>
        </Suspense>
    )
);