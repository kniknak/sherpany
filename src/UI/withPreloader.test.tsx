import React from "react";
import { shallow } from "enzyme";
import { withPreloader } from "./withPreloader";

describe("Preloader", () => {
    describe("withPreloader", () => {
        it("renders", () => {
            const Foo = () => null
            const WithPreloader = withPreloader(Foo)
            const subject = shallow(<WithPreloader/>);

            expect(subject).toMatchSnapshot();
        });
    });
});
