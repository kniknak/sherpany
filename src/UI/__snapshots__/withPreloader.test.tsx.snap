// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader withPreloader renders 1`] = `
<Suspense
  fallback={
    <UNDEFINED>
      loading...
    </UNDEFINED>
  }
>
  <Foo />
</Suspense>
`;
